core/modules/css/CssAccess.cc

```diff
@@ -0,0 +1,848 @@
+        // check table key
+        if (not _kvI->exists(tableKey)) throw NoSuchTable(dbName, tableName);
+    }
+    return schema;
+}
+
+MatchTableParams
+CssAccess::getMatchTableParams(std::string const& dbName,
+                               std::string const& tableName) const {
+    LOGF(_log, LOG_LVL_DEBUG, "getMatchTableParams(%1%, %2%)" % dbName % tableName);
+
+    std::string const tableKey = _prefix + "/DBS/" + dbName + "/TABLES/" + tableName;
+
+    MatchTableParams params;
+
+    std::vector<std::string> subKeys{"dirTable1", "dirColName1", "dirTable2", "dirColName2", "flagColName"};
+    auto paramMap = _getSubkeys(tableKey + "/match", subKeys);
+    if (paramMap.empty()) {
+        // check table key
+        if (not _kvI->exists(tableKey)) throw NoSuchTable(dbName, tableName);
+        return params;
+    }
+
+    // fill the structure
+    params.dirTable1 = paramMap["dirTable1"];
+    params.dirColName1 = paramMap["dirColName1"];
+    params.dirTable2 = paramMap["dirTable2"];
+    params.dirColName2 = paramMap["dirColName2"];
+    params.flagColName = paramMap["flagColName"];
+
+    return params;
+}
+
+PartTableParams
+CssAccess::getPartTableParams(std::string const& dbName,
+                              std::string const& tableName) const {
+    LOGF(_log, LOG_LVL_DEBUG, "getPartTableParams(%1%, %2%)" % dbName % tableName);
+
+    std::string const tableKey = _prefix + "/DBS/" + dbName + "/TABLES/" + tableName;
+
+    PartTableParams params;
+
+    std::vector<std::string> subKeys{"partitioning", "partitioning.json", "partitioning/subChunks",
+        "partitioning/dirDb", "partitioning/dirTable", "partitioning/dirColName", "partitioning/latColName",
+        "partitioning/lonColName", "partitioning/overlap", "partitioning/secIndexColName"};
+    auto paramMap = _getSubkeys(tableKey, subKeys);
+    if (paramMap.empty()) {
+        // check table key
+        if (not _kvI->exists(tableKey)) throw NoSuchTable(dbName, tableName);
+        return params;
+    }
+
+    // fill the structure
+    params.dirDb = paramMap["partitioning/dirDb"];
+    params.dirTable = paramMap["partitioning/dirTable"];
+    params.dirColName = paramMap["partitioning/dirColName"];
+    params.latColName = paramMap["partitioning/latColName"];
+    params.lonColName = paramMap["partitioning/lonColName"];
+    params.partitioned =
+            paramMap.count("partitioning") + paramMap.count("partitioning.json") > 0;
+    try {
+        auto iter = paramMap.find("partitioning/subChunks");
+        if (iter != paramMap.end()) {
+            params.subChunks = std::stoi(iter->second);
+        }
+        iter = paramMap.find("partitioning/overlap");
+        if (iter != paramMap.end()) {
+            params.overlap = std::stod(iter->second);
+        }
+    } catch (std::exception const& exc) {
+        throw KeyValueError(tableKey + "/partitioning", "one of the sub-keys is not numeric: " +
+                            std::string(exc.what()));
```

Comment:
I added ERROR message in addition to exception (indeed exception does not have full info, logging should help)